src/services/shop/shop-router.ts

```diff
@@ -0,0 +1,265 @@
+ *      "quantity": 1
+ *   }]
+ * }
+ *
+ * @apiUse weakJwtVerification
+ * */
+shopRouter.get("/", weakJwtVerification, async (_: Request, res: Response, next: NextFunction) => {
+    try {
+        const shopItems: ShopItemFormat[] = await Models.ShopItem.find();
+        const shopQuantities: QuantityFormat[] = await Models.ShopQuantity.find();
+
+        const itemsMap = new Map<string, ShopItemFormat>();
+        shopItems.forEach((item) => {
+            return itemsMap.set(item.itemId, item);
+        });
+
+        const itemsWithQuantity: ItemFormat[] = Array.from(itemsMap.values()).map((item) => {
+            const quantity = shopQuantities.find((q) => {
+                return q.itemId === item.itemId;
+            });
+            const itemQuantity = quantity ? quantity.quantity : 0;
+
+            return {
+                itemId: item.itemId,
+                name: item.name || "",
+                price: item.price || 0,
+                isRaffle: item.isRaffle || false,
+                quantity: itemQuantity,
+            };
+        });
+
+        return res.status(StatusCode.SuccessOK).send(itemsWithQuantity);
+    } catch (error) {
+        console.error(error);
+        return next(new RouterError(StatusCode.ServerErrorInternal, "InternalServerError"));
+    }
+});
+
+/**
+ * @api {post} /shop/ POST /shop/
+ * @apiGroup Shop
+ * @apiDescription Create a new item.
+ *
+ * @apiHeader {String} Authorization User's JWT Token with admin permissions.
+ *
+ * @apiBody {Json} sitem The item details to be created.
+ * @apiParamExample {Json} Request Body Example for Shop Item:
+ * {
+ *   "name": "hoodie",
+ *   "price": 15,
+ *   "isRaffle": true
+ *   "quantity": 20
+ * }
+ *
+ * @apiSuccess (201: Created) {Json} item The created item details.
+ * @apiSuccessExample Example Success Response for Shop Item
+ * HTTP/1.1 201 Created
+ * {
+ *   "item": {
+ *     "id": "52fdfc072182654f163f5f0f9a621d72",
+ *     "name": "hoodie",
+ *     "price": 15,
+ *     "isRaffle": true
+ *   }
+ *  "quantity": {
+ *      "id": "52fdfc072182654f163f5f0f9a621d72",
+ *      "quantity:" 20
+ *   }
+ * }
+ *
+ * @apiUse strongVerifyErrors
+ * @apiError (400: Bad Request) {String} InvalidParams Invalid item parameters provided.
+ * @apiError (403: Forbidden) {String} InvalidPermission User does not have admin permissions.
+ * @apiError (500: Internal Server Error) {String} InternalError An error occurred on the server.
+ */
+shopRouter.post("/", strongJwtVerification, async (req: Request, res: Response, next: NextFunction) => {
+    const payload: JwtPayload = res.locals.payload as JwtPayload;
+    // Check if the token has staff permissions
+    if (!hasStaffPerms(payload)) {
+        return next(new RouterError(StatusCode.ClientErrorForbidden, "InvalidPermission"));
+    }
+
+    // Convert item request into the item format
+    const itemFormatString: string = req.body;
+    const itemFormat: ItemFormat = JSON.parse(itemFormatString);
```

Comment:
this could be problematic? why not follow how other services are being parsed ([example](https://github.com/HackIllinois/adonix/blob/main/src/services/event/event-router.ts#L364))